bin/check_pylint_diff

```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/env bash
+# Copyright (C) 2016 Google Inc., authors, and contributors <see AUTHORS file>
+# Licensed under http://www.apache.org/licenses/LICENSE-2.0 <see LICENSE file>
+# Created By: miha@reciprocitylabs.com
+# Maintained By: miha@reciprocitylabs.com
+
+set -o nounset
+set -o errexit
+
+ARG1=${1:-}
+TMP_REPO=".temp_repo"
+SCRIPT=$(basename "$0")
+SCRIPTPATH=$( cd "$(dirname "$0")" ; pwd -P )
+
+cd "${SCRIPTPATH}/.."
+
+
+if [[ "$ARG1" == "-h" || "$ARG1" == "--help" ]]; then
+  echo "
+Usage: $SCRIPT [test_commit] [-h]
+
+This script will compare pylint error count from two different commits.
+Note: all changes that are not committed will be ignored.
+
+
+Given the commit tree:
+
+       D---E---F---G---H
+            \\         /
+             A---B---C
+
+- Running '$SCRIPT' on C will check the diff between B and C
+- Running '$SCRIPT' on H will check the diff between G and H
+- Running '$SCRIPT F' on H will check the diff between F and H
+"
+  exit 0
+fi
+
+rm -rf $TMP_REPO
+mkdir -p $TMP_REPO
+cp -a .git $TMP_REPO/
+cd $TMP_REPO
+git reset --hard HEAD
+
+TEST_COMMIT=$(git rev-parse HEAD)
+if [[ "$ARG1" != "" ]]; then
+  PARENT_1=$ARG1
+else
+  PARENT_1=$(git show --pretty=raw $TEST_COMMIT | grep parent | head -1 | grep -oE "[^ ]*$")
```

Comment:
This could be `git show --pretty=raw $TEST_COMMIT | awk '/^parent/{ print $2 }'` for a bit simpler.